core/chains/evm/gas/suggested_price_estimator.go

```diff
@@ -31,17 +33,12 @@ type suggestedPriceConfig interface {
 	BumpMin() *assets.Wei
 }
 
-//go:generate mockery --quiet --name rpcClient --output ./mocks/ --case=underscore --structname RPCClient
-type rpcClient interface {
-	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
-}
-
 // SuggestedPriceEstimator is an Estimator which uses the suggested gas price from eth_gasPrice.
 type SuggestedPriceEstimator struct {
 	services.StateMachine
 
 	cfg        suggestedPriceConfig
-	client     rpcClient
+	client     ethClient
```

Comment:
If I understood Dimitris suggestion for go best practices, maybe we want to keep a local client here but make it component specific like `suggestedPriceEstimatorClient`. Then we can use that in the struct and builder below so we really drill down that this component only needs `CallContext`.

Reply:
I added this back in as you suggested, but kept the client in the `NewSuggestedPriceEstimator` method signature parameters as `feeEstimatorClient` since that's what `NewL1GasOracle` requires.